awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -260,13 +260,21 @@ def make_request(self, method, endpoint, *args, **kwargs):
                 tower_type = response.info().getheader('X-API-Product-Name', None)
                 tower_version = response.info().getheader('X-API-Product-Version', None)
 
-            collection_major = Version(self._COLLECTION_VERSION).version[0]
-            tower_major = Version(tower_version).version[0]
+            parsed_collection_version = Version(self._COLLECTION_VERSION).version
+            parsed_tower_version = Version(tower_version).version
+            if tower_type != 'AWX':
+                collection_compare_ver = parsed_collection_version[0]
+                tower_compare_ver = parsed_tower_version[0]
+            else:
+                collection_compare_ver = "{}.{}".format(parsed_collection_version[0], parsed_collection_version[1])
+                tower_major = '{}.{}'.format(parsed_tower_version[0], parsed_tower_version[1])
```

Comment:
`tower_major` needs to be `tower_compare_ver`, I think

Reply:
Otherwise I get a "variable was referenced before assignment" wall o' red when running tests